lib/agent/meson.build

```diff
@@ -28,13 +28,19 @@ if host_os_family != 'darwin'
   operations += ['--move', 'destructor', 'frida_deinit_memory', 'first']
 endif
 
+if host_machine.endian() == 'little'
+  endian_option = ['--endian', 'little']
+else
+  endian_option = ['--endian', 'big']
+endif
```

Comment:
Can't we detect this by looking at the `ELF` like we're already doing in modulate.py? (We're already sniffing it to figure out the architecture type.)

Reply:
I did have a look at this, but it seems that objdump doesn't explicitly list the endianness. It instead list the architecture, but isn't very consistent in the naming. It seems different architectures has very different output. Also I wasn't sure how stable those strings would be between different versions of objdump. Happy to change to a whitelist type approach if you think it will work though?

Otherwise, I though of looking at using pyelftools to parse the elf properly, but this would add a dependency.

test-armbe8:     file format elf32-bigarm
architecture: armv6, flags 0x00000112:
EXEC_P, HAS_SYMS, D_PAGED
start address 0x0001070c

/test-arm:     file format elf32-littlearm
architecture: armv5t, flags 0x00000112:
EXEC_P, HAS_SYMS, D_PAGED
start address 0x00010698

/bin/ls:     file format elf64-x86-64
architecture: i386:x86-64, flags 0x00000150:
HAS_SYMS, DYNAMIC, D_PAGED
start address 0x0000000000005850

Reply:
Ah ok, let's do it this way for now.